components/core/src/compressor_frontend/InputBuffer.cpp

```diff
@@ -0,0 +1,111 @@
+// C++ libraries
+#include <memory.h>
+#include <string>
+
+// spdlog
+#include <spdlog/spdlog.h>
+
+// Project Headers
+#include "InputBuffer.hpp"
+
+using std::string;
+using std::to_string;
+
+namespace compressor_frontend {
+
+    void InputBuffer::reset () {
+        m_at_end_of_file = false;
+        m_finished_reading_file = false;
+        m_consumed_pos = 0;
+        m_bytes_read = 0;
+        m_last_read_first_half = false;
+        Buffer::reset();
+    }
+
+    bool InputBuffer::read_is_safe () {
+        if (m_finished_reading_file) {
+            return false;
+        }
+        // If the next message starts at 0, the previous character is at m_curr_storage_size - 1
+        if (m_consumed_pos == -1) {
+            m_consumed_pos = m_curr_storage_size - 1;
+        }
+        // Check that the last log message ends in the half of the buffer that was last read.
+        // This means the other half of the buffer has already been fully used.
+        if ((!m_last_read_first_half && m_consumed_pos > m_curr_storage_size / 2) ||
```

Comment:
* Rather than using `m_curr_storage_size / 2` everywhere, it seems to me like it would more concise to have a variable like `m_single_buffer_size` and then in the few places you need the double buffer size, you can use `m_single_buffer_size * 2`.
* Since you return true if this is true and false if it's not, you can just use expression as the return value.